src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -193,11 +195,14 @@ module.exports = function<T, P, I, TI, C>(
         commitNestedUnmounts(node);
         // After all the children have unmounted, it is now safe to remove the
         // node from the tree.
-        if (parent) {
+        if (parent && !ignoreHostNode) {
           removeChild(parent, node.stateNode);
         }
       } else {
-        commitUnmount(node);
+        // If ignoreTop is true, don't unmount the top fiber
+        if (!ignoreTop || node !== current) {
```

Comment:
Maybe it would be easier to understand if the comment emphasized when this happens (e.g. "If node and current are different, it means X. When `ignoreTop` is set, it means Y. Therefore we do Z to implement this [scenario described from user's point of view].")